File: old/rin.Framework/Graphics/Windows/Events/KeyEvent.cs
namespace rin.Framework.Graphics.Windows.Events;

public class KeyEvent : Event
{
    public required InputKey Key;
    public required InputState State;
    public required InputModifier Modifiers;
    public bool IsAltDown => Modifiers.HasFlag(InputModifier.Alt);
    public bool IsControlDown => Modifiers.HasFlag(InputModifier.Control);
    public bool IsShiftDown => Modifiers.HasFlag(InputModifier.Shift);
}

File: lib/rin/io/events/KeyEvent.cpp
#include "rin/io/events/KeyEvent.h"
namespace rin::io
{


    KeyEvent::KeyEvent(Window* inWindow, const InputKey& inKey, const InputState& inState, const Flags<InputModifier>& inModifiers) : Event(inWindow), key(inKey), state(inState),
        modifiers(inModifiers)
    {

    }
}
